public/app/features/explore/Logs/LogsMetaRow.tsx

```diff
@@ -88,9 +88,39 @@ export const LogsMetaRow = memo(
           break;
         case DownloadFormat.CSV:
           const dataFrameMap = new Map<string, DataFrame>();
+
+          let sortedValue = new Map<string, any | undefined[]>();
+          logRows.forEach((row) => {
+            if (row.dataFrame?.refId) {
+              // handle Time and Line
+              const timeKey = row.dataFrame.refId + 'Time';
+              const lineKey = row.dataFrame.refId + 'Line';
+              const labelKey = row.dataFrame.refId + 'labels';
+              if (!sortedValue.has(timeKey)) {
+                sortedValue.set(timeKey, []);
+              }
+              if (!sortedValue.has(lineKey)) {
+                sortedValue.set(lineKey, []);
+              }
+              if (!sortedValue.has(labelKey)) {
+                sortedValue.set(labelKey, []);
+              }
+              sortedValue.get(timeKey).push(row.timeEpochMs);
+              sortedValue.get(lineKey).push(row.entry);
+              sortedValue.get(labelKey).push(row.labels);
+            }
+          });
+
           logRows.forEach((row) => {
+            // reorder dataFrame
             if (row.dataFrame?.refId && !dataFrameMap.has(row.dataFrame?.refId)) {
-              dataFrameMap.set(row.dataFrame?.refId, row.dataFrame);
+              for (let fieldIdx in row.dataFrame.fields) {
+                const key = row.dataFrame.refId + row.dataFrame.fields[fieldIdx].name;
+                if (sortedValue.has(key)) {
+                  row.dataFrame.fields[fieldIdx].values = sortedValue.get(key);
+                }
+              }
+              dataFrameMap.set(row.dataFrame.refId, row.dataFrame);
             }
           });
           dataFrameMap.forEach(async (dataFrame) => {
```

Comment:
The expected outcome of this PR would be to generate the CSV files with the same visible order that the user is seeing in the log list. I would recommend reverting this changes and:

- Passing the logs sort order prop to this component.
- Trying sorted the downloaded values with a transformation, like:

```
{
              "id": "sortBy",
              "options": {
                "fields": {},
                "sort": [
                  {
                    "desc": true, // depending on logs sort order
                    "field": "timestamp" // the field can be timestamp or time. You should loop over the fields and find fields with type time
                  }
                ]
              }
            }
```

And check that the generated field has the same visible order than the list in Explore.